include/deal.II/fe/fe_values.h

```diff
@@ -1764,10 +1764,10 @@ class FEValuesBase : protected FEValuesData<dim,spacedim>,
    *
    * @dealiiRequiresUpdateFlags{update_values}
    */
-  template <class InputVector>
+  template <class InputVector, typename number>
   void get_function_values (const InputVector &fe_function,
                             const VectorSlice<const std::vector<types::global_dof_index> > &indices,
-                            VectorSlice<std::vector<std::vector<double> > > values,
+                            VectorSlice<std::vector<std::vector<number> > > values,
```

Comment:
Here and in the functions below, I think the better way (other than adding a second template argument, with all of the headaches that presents) would actually be to write it like this:
```
template <class InputVector>
void get_function_values (const InputVector &fe_function,
                     const VectorSlice<const std::vector<types::global_dof_index> > &indices,
                     VectorSlice<std::vector<std::vector<typename InputVector::value_type> > > values, ...)
```
In other words, use `typename InputVector::value_type` as the scalar type for the inner vector. Would you mind reworking the patch in this direction?